qtcodes/circuits/repetition.py

```diff
@@ -1,438 +1,63 @@
-        """
-        chunks = readout_string.split(" ")
-
-        if len(chunks[0]) > 1:  # this is true when all data qubits are readout
-            assert (
-                readout_type == "Z"
-            ), "Rep code currently only supports Z lattice readout."
-            (
-                logical_readout,
-                final_stabilizer,
-            ) = self.extract_final_stabilizer_and_logical_readout_z(chunks[0])
-            chunks = [
-                final_stabilizer,
-            ] + chunks[1:]
-        else:
-            logical_readout = int(chunks[0])
-            chunks = chunks[1:]
-
-        int_syndromes = [int(x, base=2) for x in chunks[::-1]]
-        z_syndromes = [a ^ b for (a, b) in zip(int_syndromes, int_syndromes[1:])]
-
-        Z = []
-        for T, syndrome in enumerate(z_syndromes):
-            for loc in range(int(self.params["num_syn"])):
-                if syndrome & 1 << loc:
-                    Z.append((float(T), 0.5 + loc, 0.0))
-        return (
-            logical_readout,
-            {"Z": Z},
-        )
-
-
-class RepetitionQubit(TopologicalQubit):
+class RepetitionQubit(XXZZQubit):
     """
     A single logical repetition code qubit. At the physical level, this wraps a
     circuit, so we chose to subclass and extend TopologicalQubit which extends QuantumCircuit.
     """
 
-    lattice_type = _RepetitionLattice
-
-    def stabilize(self) -> None:
-        """
-        Run a single round of stabilization (entangle and measure).
-        """
-        self.lattice.params["T"] += 1
-        syndrome_readouts = ClassicalRegister(
-            self.lattice.params["num_syn"],
-            name=self.name + "_c{}".format(self.lattice.params["T"]),
-        )
-        self.lattice.cregisters[
-            "syndrome{}".format(self.lattice.params["T"])
-        ] = syndrome_readouts
-        self.circ.add_register(syndrome_readouts)
-
-        self.lattice.entangle()
+    def __init__(
+        self,
+        params: Optional[Dict[str, int]] = None,
+        name: str = "tq",
+        circ: Optional[QuantumCircuit] = None,
+    ) -> None:
+        params = params if params else {}
+
+        if "phase-flip-protected" not in params:
+            params["phase-flip-protected"] = False
+
+        # The repetition code is a 1D lattice
+        if "d" not in params:
+            if params["phase-flip-protected"]:
+                params["d"] = (1, 3)
+            else:
+                params["d"] = (3, 1)
+        elif isinstance(params["d"], Number):
+            d = int(params["d"])
+            if params["phase-flip-protected"]:
+                params["d"] = (1, d)
+            else:
+                params["d"] = (d, 1)
+        elif isinstance(params["d"], Tuple):
+            if not params["phase-flip-protected"] and params["d"][constants.DW] != 1:
+                raise LatticeError(
+                    "Repetition qubits can only have width 1 in parameter d: e.g. (3,1)."
```

Comment:
Perhaps change this to:
"Bit-flip protected repetition qubits can only have width 1 in parameter d: e.g. (3,1). If you intend to create a phase-flip protected repetition qubit (e.g. with d=(1,3)), then please set the phase-flip-protected flag parameter to True."